Mathlib/Data/Matroid/Minor/Basic.lean

```diff
@@ -716,4 +716,197 @@ lemma IsCocircuit.delete_diff_isCocircuit {X : Set α} (hK : M.IsCocircuit K) (h
+  rw [contract_restrict_eq_restrict_contract _ disjoint_sdiff_right]
+  simp [union_eq_self_of_subset_right hCR]
+
+/-- Contraction and deletion commute for disjoint sets. -/
+lemma contract_delete_comm (M : Matroid α) (hCD : Disjoint C D) : M ／ C ＼ D = M ＼ D ／ C := by
+  wlog hCE : C ⊆ M.E generalizing C with aux
+  · rw [← contract_inter_ground_eq, aux (hCD.mono_left inter_subset_left) inter_subset_right,
+      contract_eq_contract_iff, inter_assoc, delete_ground,
+      inter_eq_self_of_subset_right diff_subset]
+  rw [delete_eq_restrict, delete_eq_restrict, contract_ground, diff_diff_comm,
+    restrict_contract_eq_contract_restrict _ (by simpa [hCE, subset_diff])]
+
+/-- A version of `contract_delete_comm` without the disjointness hypothesis,
+and hence a less simple RHS. -/
+lemma contract_delete_comm' (M : Matroid α) (C D : Set α) : M ／ C ＼ D = M ＼ (D \ C) ／ C := by
+  rw [contract_delete_diff, contract_delete_comm _ disjoint_sdiff_right]
+
+lemma delete_contract_eq_diff (M : Matroid α) (D C : Set α) : M ＼ D ／ C = M ＼ D ／ (C \ D) := by
+  rw [contract_eq_contract_iff, delete_ground, ← diff_inter_distrib_right, diff_eq, diff_eq,
+    inter_assoc]
+
+/-- A version of `delete_contract_comm'` without the disjointness hypothesis,
+and hence a less simple RHS. -/
+lemma delete_contract_comm' (M : Matroid α) (D C : Set α) : M ＼ D ／ C = M ／ (C \ D) ＼ D := by
+  rw [delete_contract_eq_diff, ← contract_delete_comm _ disjoint_sdiff_left]
+
+/-- A version of `contract_delete_contract` without the disjointness hypothesis,
+and hence a less simple RHS. -/
+lemma contract_delete_contract' (M : Matroid α) (C D C' : Set α) :
+    M ／ C ＼ D ／ C' = M ／ (C ∪ C' \ D) ＼ D := by
+  rw [delete_contract_eq_diff, ← contract_delete_comm _ disjoint_sdiff_left, contract_contract]
+
+lemma contract_delete_contract (M : Matroid α) (C D C' : Set α) (h : Disjoint C' D) :
+    M ／ C ＼ D ／ C' = M ／ (C ∪ C') ＼ D := by rw [contract_delete_contract', sdiff_eq_left.mpr h]
+
+/-- A version of `contract_delete_contract_delete` without the disjointness hypothesis,
+and hence a less simple RHS. -/
+lemma contract_delete_contract_delete' (M : Matroid α) (C D C' D' : Set α) :
+    M ／ C ＼ D ／ C' ＼ D' = M ／ (C ∪ C' \ D) ＼ (D ∪ D') := by
+  rw [contract_delete_contract', delete_delete]
+
+lemma contract_delete_contract_delete (M : Matroid α) (C D C' D' : Set α) (h : Disjoint C' D) :
+    M ／ C ＼ D ／ C' ＼ D' = M ／ (C ∪ C') ＼ (D ∪ D') := by
+  rw [contract_delete_contract_delete', sdiff_eq_left.mpr h]
+
+/-- A version of `delete_contract_delete` without the disjointness hypothesis,
+and hence a less simple RHS. -/
+lemma delete_contract_delete' (M : Matroid α) (D C D' : Set α) :
+    M ＼ D ／ C ＼ D' = M ／ (C \ D) ＼ (D ∪ D') := by
+  rw [delete_contract_comm', delete_delete]
+
+lemma delete_contract_delete (M : Matroid α) (D C D' : Set α) (h : Disjoint C D) :
+    M ＼ D ／ C ＼ D' = M ／ C ＼ (D ∪ D') := by
+  rw [delete_contract_delete', sdiff_eq_left.mpr h]
+
+/-! ### Minors -/
```

Comment:
Certainly everything above this line looks ready for insta-merging.

Also, I'm wondering if the first ~800 lines of this file should belong in a more prerequisite file. So that `Minor/Basic.lean` can really start with the definition of minors.

Reply:
This could make sense. What would you think about `Matroid.Minor.Delete` and `Matroid.Minor.Contract` being separate files, each about just one definition? And then `Matroid.Minor.Order` for `IsMinor`?

Reply:
Potentially the current 'Restriction' file could also go in the `Minor` subdirectory, even though it is way further back in the import heirarchy.

Reply:
Ok, I've just done a reshuffle to this effect - now the `Minor` folder has `Contract`/`Delete`/`Restrict`/`Order` as different files.